gtsfm/frontend/detector_descriptor/superpoint.py

```diff
@@ -73,3 +75,14 @@ def detect_and_describe(self, image: Image) -> Tuple[Keypoints, np.ndarray]:
         keypoints = Keypoints(feature_points, responses=scores, scales=np.ones(scores.shape))
 
         return keypoints, descriptors
+
+    def get_delayed_extra_args(self) -> Delayed:
+        if self._extra_args is None:
+            model = SuperPoint(self._config).cpu()
```

Comment:
Same question as John. Why force CPU instead of checking if `torch.cuda_is_available()`?

Reply:
Because we are using the delayed version of the model. We cannot map the model to a GPU and pass it around to different workers. If GPU processing is to be done, we need to do the mapping on the worker code,  i.e. in the `detect_and_describe_function`.